pest/src/error.rs

```diff
@@ -25,7 +24,9 @@ use crate::RuleType;
 
 /// Parse-related error type.
 #[derive(Clone, Debug, Eq, Hash, PartialEq)]
-pub struct Error<R> {
+#[cfg_attr(feature = "std", derive(thiserror::Error))]
+#[cfg_attr(feature = "std", error("An error was encountered while parsing:\n{}", self.format()))]
```

Comment:
Note: behavior change

Reply:
I believe the standard convention starts with a lowercase character, as to fit in a display of `Error: {your_error}`. See e.g. [regex's syntax error](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ecd77467077c618713db50b686ff4011) ([multiline](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=cdc148b1b58aab15ec7e14ce206e0a03)).

To match existing convention, I think this would be `"parse error:\n{format}"`.

Reply:
In fact, I think this should just be `error("{}", self.format())`, because it won't change behaviour.